### DISCUSSION TITLE
yieldday um Mitternacht auf 0 zurücksetzen

### DISCUSSION THREAD

**Discussion by UweHM** [tbnobody/OpenDTU#182] [Ideas]
Ich habe die SW auf einem ESP32 mit NRF24L01+ ohne jegliche Schwierigkeiten zum laufen gebracht.
Das war genial einfach und funktioniert super! - Tolle Arbeit von Euch!

Ich habe die Daten über einen MQTT-Server nun in FHEM eingebunden und lasse mir die Daten in "schönen" Diagrammen anzeigen.
Wäre es möglich, die Werte von yieldday automatisch beim Übergang auf den nächsten Tag zu 0 zu setzen?
Der Umrichter schaltet ja bei zunehmender Dunkelheit ab. Die Werte von yieldday bleiben aber nach Mitternacht noch erhalten, bis er wieder einschaltet.

**Comment by viktorhotel** [comment]
Hallo zusammen,
würde dies auch wünschen:
![grafik](https://user-images.githubusercontent.com/6044476/204095397-03b19cb6-4048-4bae-bcd8-782dd0481fb3.png)

Leider ist es so, das yieldday vom inverter erst zurückgesetzt wird, wenn er neu anfängt zu produzieren.  Dies verfälscht natürlich die Werte vom nächsten Tag (von MIDNIGHT bis Inverterstart). Leider arbeitet hier die Hoymiles Firmware unsauber. Ich könnte mir vorstellen, das opendtu diesen Fehler "ausgebügeln" könnte. 
edit:
Die Aussage mit der unsauberen Firmware des Hoymiles ist natürlich falsch:  Der Inverter kann ja nicht anders. Er ist zu Midnight in Standby und sendet erst wieder wenn er produziert.

**Comment by viktorhotel** [comment]
Im Moment löse ich das Problem so (openhab und MQTT).
Nach Abschalten des Inverters werden keine MQTT Daten von opendtu gesendet.
Ich sende nun per publish um 0.00 Uhr eine "0"
Die Rule in openhab sieht dann so aus (die Inverter serial Nr ist nätürlich ohne xxxxx):
rule "opendtu YieldDay um Mitternacht auf 0 setzen"
when
   Time cron "0 0 0 ? * * *"       //At 00:00:00am every day 
   //Time cron "0 * * * * ?"        //for debug At second 0 of every minute
then
 val mqttActions = getActions("mqtt","mqtt:broker:27fa6bea08")
 mqttActions.publishMQTT("solar/1141xxxxxxxxx/0/yieldday","0",true)
end  

![grafik](https://user-images.githubusercontent.com/6044476/204129543-6cdad292-0a80-4033-a27d-8b4b23002788.png)

**Reply by frankh-gituser** [reply]
habe das gerade auch probiert. die 0 wird zwar auf das topic geschrieben - aber der alte Wert bleibt existent. Erst beim restart des HM wird der Wert wieder auf 0 gesetzt. Auch wenn ich den HM ausschalte und dann den Wert auf 0 setze, nimmt er den Wert nicht an. Auf den cmd topics für die limit Einstellungen nimmt er die Werte an. Irgendeine Idee ?

**Comment by viktorhotel** [comment]
Nochmal ein kleines Update: Aufgrund der Problematik, das Yieldday tagsüber mal auf 0 gehen kann (beim Inverter Neustart) mache ich es jetzt anders: Ich berechne die Differenz von Yieldtotal um 0.01 Uhr bis aktuell.
Im Grunde ist das einfacher als die o.a. Umgehung.
